chihuahua/src/hcall/common.rs

```diff
@@ -364,16 +341,40 @@ impl<Module, Memory> HostProvisioningState<Module, Memory> {
         self.result = Some(result.to_vec());
     }
 
+    /// Registers the previous result.
+    /// If the previous computation does not produce any result, ie returning none,
+    /// it is converted to an empty vector.
+    /// This distinguishes from `None`, which indicates that it is the first round of computation.
+    #[inline]
+    pub(crate) fn set_previous_result(&mut self, result: &Option<Vec<u8>>) {
+        self.previous_result = Some(
+            result
+                .as_ref()
+                .unwrap_or(&vec![])
+                //.unwrap_or(&pinecone::to_vec::<Vec<u8>>(&vec![]).unwrap())
```

Comment:
We shouldn't commit dead or commented-out code into the main branch.  Can this line be removed?